boa_engine/src/builtins/temporal/mod.rs

```diff
@@ -0,0 +1,1149 @@
+//! The ECMAScript `Temporal` stage 3 built-in implementation.
+//!
+//! More information:
+//!
+//! [spec]: https://tc39.es/proposal-temporal/
+#![allow(unreachable_code, dead_code, unused_imports)] // Unimplemented
+
+mod calendar;
+mod date_equations;
+mod duration;
+mod fields;
+mod instant;
+mod now;
+mod plain_date;
+mod plain_date_time;
+mod plain_month_day;
+mod plain_time;
+mod plain_year_month;
+mod time_zone;
+mod zoned_date_time;
+
+use std::ops::Mul;
+
+#[cfg(feature = "experimental")]
+#[cfg(test)]
+mod tests;
+
+pub(crate) use fields::TemporalFields;
+
+use self::date_equations::mathematical_days_in_year;
+pub use self::{
+    calendar::*, duration::*, instant::*, now::*, plain_date::*, plain_date_time::*,
+    plain_month_day::*, plain_time::*, plain_year_month::*, time_zone::*, zoned_date_time::*,
+};
+use super::{BuiltInBuilder, BuiltInObject, IntrinsicObject};
+use crate::{
+    builtins::iterable::IteratorRecord,
+    context::intrinsics::{Intrinsics, StandardConstructors},
+    js_string,
+    object::{internal_methods::get_prototype_from_constructor, ObjectData, ObjectInitializer},
+    property::{Attribute, PropertyKey},
+    realm::Realm,
+    string::utf16,
+    value::{IntegerOrInfinity, Type},
+    Context, JsBigInt, JsNativeError, JsNativeErrorKind, JsObject, JsResult, JsString, JsSymbol,
+    JsValue, NativeFunction,
+};
+use boa_ast::temporal::{self, UtcOffset};
+use boa_profiler::Profiler;
+
+// Relavant numeric constants
+pub(crate) const NS_PER_DAY: i64 = 86_400_000_000_000;
+pub(crate) const MICRO_PER_DAY: i64 = 8_640_000_000;
+pub(crate) const MILLI_PER_DAY: i64 = 8_600_000;
```

Comment:
```suggestion
pub(crate) const NANOS_PER_DAY: i64 = MICROS_PER_DAY * 1000;
pub(crate) const MICROS_PER_DAY: i64 = MILLIS_PER_DAY * 1000;
pub(crate) const MILLIS_PER_DAY: i64 = 24 * 60 * 60 * 1000; // 24 hours * 60 minutes * 60 seconds * 1000 millis
```

`MILLIS_PER_DAY` is wrong. We should probably use simple math to avoid calculation errors like this.

Reply:
😱 I missed the 4!!!